crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,217 @@
+        return true;
+    }
+    // VSCode temporary files
+    if std::path::Path::new(file_name)
+        .extension()
+        .map_or(false, |ext| ext.eq_ignore_ascii_case("tmp"))
+    {
+        return true;
+    }
+
+    // Add more patterns for other editors as needed
+
+    false
+}
+
+impl Cmd {
+    pub async fn run(&mut self) -> Result<(), Error> {
+        let (tx, mut rx) = mpsc::channel::<Message>(100);
+        let rebuild_state = Arc::new(Mutex::new(false));
+        let workspace_root: &Path = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .unwrap_or_else(|| Path::new("."));
+        let env_toml_path = Arc::new(workspace_root.join("environments.toml"));
+        let env_toml_parent = Arc::new(
+            env_toml_path
+                .parent()
+                .unwrap_or(Path::new(""))
+                .to_path_buf(),
+        );
+
+        let mut watched_dirs = Vec::new();
+        let packages = self
+            .build_cmd
+            .list_packages()?
+            .into_iter()
+            .map(|package| {
+                Arc::new(PathBuf::from(
+                    package.manifest_path.parent().unwrap().as_str(),
+                ))
+            })
+            .collect::<Vec<_>>();
+
+        for package_path in &packages {
+            watched_dirs.push(package_path.clone());
+            eprintln!("Watching {}", package_path.as_path().display());
+        }
+        let watched_dirs_clone = watched_dirs.clone();
+        let env_toml_path_clone = Arc::clone(&env_toml_path);
+        let env_toml_parent_clone = Arc::clone(&env_toml_parent);
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if matches!(
+                        event.kind,
+                        notify::EventKind::Create(_)
+                            | notify::EventKind::Modify(_)
+                            | notify::EventKind::Remove(_)
+                    ) {
+                        if let Some(path) = event.paths.first() {
+                            // Ignore temporary files
+                            if is_temporary_file(path) {
+                                return;
+                            }
+                            let env_toml_parent_abs = canonicalize_path(&env_toml_parent_clone);
+                            let env_toml_path_abs = canonicalize_path(&env_toml_path_clone);
+                            let parent_is_env_toml_parent =
+                                path.parent() == Some(env_toml_parent_abs.as_path());
+                            let path_is_env_toml = path == env_toml_path_abs.as_path();
+                            let parent_is_in_watched_dirs = is_parent_in_watched_dirs(
+                                &env_toml_parent_abs,
+                                &watched_dirs_clone,
```

Comment:
This seems like it could also be computed outside the closure.  I'm I don't understand why this is an issue.

Reply:
Fixed